### ISSUE TITLE
Add a `prisma.$reset()` function as shortcut to running `prisma migrate reset`

### ISSUE THREAD

**Issue by flybayer** [prisma/prisma#5230]
## Problem

`prisma migrate reset` is used heavily for testing. I want a `prisma.$reset()` function added to prisma client as super easy shortcut to reset the database.

## Suggested solution

I've added this to our Blitz adapter and it's super nice in Jest tests:

```ts
beforeEach(async () => {
  await db.$reset()
})
```

## Alternatives

None

**Comment by felinto-dev** [comment]
Is there an update from the Prisma team on the possible addition of this functionality?

I am facing the same problem and I cannot see any way, even if it is a workaround, to clean up the database (preserving the schema) between E2E tests.

**Comment by flybayer** [comment]
As a workaround, you can copy the code we use in Blitz for this: https://github.com/blitz-js/blitz/blob/canary/packages/core/src/prisma-utils.ts

Which you can then use like this:


```ts
import {enhancePrisma} from "./enhancePrisma"
import {PrismaClient} from "@prisma/client"

const EnhancedPrisma = enhancePrisma(PrismaClient)

export * from "@prisma/client"
export default new EnhancedPrisma()
```

**Comment by saevarb** [comment]
In a similar vein, it would also be nice to be able to run migrations and `db push` programmatically, for testing purposes.

**Comment by janpio** [comment]
That would be related to https://github.com/prisma/prisma/issues/4703 @saevarb

**Comment by rsaryev** [comment]
Example https://gist.github.com/rsaryev/65884be2d8992552e9a603fefabaecb3

**Comment by bonnie** [comment]
> As a workaround, you can copy the code we use in Blitz for this: https://github.com/blitz-js/blitz/blob/canary/packages/core/src/prisma-utils.ts
> 
> Which you can then use like this:
> 
> ```ts
> import {enhancePrisma} from "./enhancePrisma"
> import {PrismaClient} from "@prisma/client"
> 
> const EnhancedPrisma = enhancePrisma(PrismaClient)
> 
> export * from "@prisma/client"
> export default new EnhancedPrisma()
> ```

Looks like this has moved to https://github.com/blitz-js/blitz/blob/canary/nextjs/packages/next/stdlib/prisma-utils.ts

**Comment by rainy-me** [comment]
FYI I'm migrating from sequelize to prisma and at the moment performance using `transaction` + `executeRaw` method by  @rsaryev (https://gist.github.com/rsaryev/65884be2d8992552e9a603fefabaecb3) is close to sequelize's and much more faster (10s per reset in my jest setup) than spawning `migrate reset` cli each time .

**Comment by mbrookson** [comment]
@yue4u Agreed this would be a lot faster but it's also lacking functionality. Running a full `migrate reset` has the advantage of really resetting the db, including reapplying all migrations. Useful if there is seeded data in migration that are required for the application to function, and therefore required for tests to pass. Generally for e2e tests I'd prefer quality and confidence over speed (within reason of course).

**Comment by Ashlayyy** [comment]
Is there any indication that this will be added in the near future? It would be helpful in multiple projects of me.
